### ISSUE TITLE
[StructuralMechanicsApplication][ResponseFunctions] Avoid referencing of node and element IDs in the json 

### ISSUE THREAD

**Issue by armingeiser** [KratosMultiphysics/Kratos#3492]
As mentioned by @philbucher in #3469: 

>  you should really get rid of using Ids, this is a very weak mechanism for the following reason:
> 
> * does not work if the MP is reordered (which is an option in the modelpart reading for a while already)
> * accessing sth by Id is slow bcs it is a "find"
> * blocker for MPI
> 
> Plus, how do you get/select the node-id? isn't this cumbersome too?
> 
> I don't really have a good suggestion on how to improve this (Maybe using coordinates...?)

The idea is to use submodel parts instead.
However this introduces a little more work for the manual creation of the input files in the first place.

**Comment by MFusseder** [comment]
Yes, it introduces a little more work, but otherwise it opens the door to expand our responses in further development (e.g. tracing more nodal displacements) Also @dbaumgaertner  will need such a submodelpart for his upcoming general stress response.

**Comment by MFusseder** [comment]
When using a submodel part within the definition of a response function there might be a problem in the ```CalculateValue```-function, because these submodel parts will belong to the adjoint model part, but the value of the response is computed based on the primal model part.

**Comment by AndreasWinterstein** [comment]
What does the "Helene Fischer" label indicate?
